File: 0495. Teemo Attacking/Solution.c
int findPoisonedDuration(int* timeSeries, int timeSeriesSize, int duration)
{
    int res = 0, last = -1;
    for (int i = 0; i < timeSeriesSize; i++) {
        res += timeSeries[i] > last ? duration : duration - last + timeSeries[i];
        last = timeSeries[i] + duration;
    }
    return res;
}

File: 0495. Teemo Attacking/Solution.java
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int res = 0, last = -1;
        for (int t : timeSeries) {
            res += t > last ? duration : t - last + duration;
            last = t + duration;
        }
        return res;
    }
}
